src/gui/plugins/component_inspector/ComponentInspector.cc

```diff
@@ -728,6 +779,60 @@ void ComponentInspector::OnPose(double _x, double _y, double _z, double _roll,
   this->dataPtr->node.Request(poseCmdService, req, cb);
 }
 
+/////////////////////////////////////////////////
+void ComponentInspector::OnLight(
+  double _rSpecular, double _gSpecular, double _bSpecular, double _aSpecular,
+  double _rDiffuse, double _gDiffuse, double _bDiffuse, double _aDiffuse,
+  double _attRange, double _attLinear, double _attConstant,
+  double _attQuadratic, bool _castShadows, double _directionX,
+  double _directionY, double _directionZ, double _innerAngle,
+  double _outerAngle, double _falloff, int _type)
+{
+  std::function<void(const ignition::msgs::Boolean &, const bool)> cb =
+      [](const ignition::msgs::Boolean &/*_rep*/, const bool _result)
+  {
+    if (!_result)
+      ignerr << "Error setting light configuration" << std::endl;
+  };
+
+  ignition::msgs::Light req;
+  req.set_name(this->dataPtr->entityName);
+  req.set_id(this->dataPtr->entity);
+  ignition::msgs::Set(req.mutable_diffuse(),
+    ignition::math::Color(_rDiffuse, _gDiffuse, _bDiffuse, _aDiffuse));
+  ignition::msgs::Set(req.mutable_specular(),
+    ignition::math::Color(_rSpecular, _gSpecular, _bSpecular, _aSpecular));
+  req.set_range(_attRange);
+  req.set_attenuation_linear(_attLinear);
+  req.set_attenuation_constant(_attConstant);
+  req.set_attenuation_quadratic(_attQuadratic);
+  req.set_cast_shadows(_castShadows);
+  if (_type == 0)
+    req.set_type(ignition::msgs::Light::POINT);
+  else if (_type == 1)
+    req.set_type(ignition::msgs::Light::SPOT);
+  else
+    req.set_type(ignition::msgs::Light::DIRECTIONAL);
+
+  if (_type == 1)  // sdf::LightType::SPOT
+  {
+    req.set_spot_inner_angle(_innerAngle);
+    req.set_spot_outer_angle(_outerAngle);
+    req.set_spot_falloff(_falloff);
+  }
+
+  // if sdf::LightType::SPOT || sdf::LightType::DIRECTIONAL
+  if (_type == 1 || _type == 2)
+  {
+    ignition::msgs::Set(req.mutable_direction(),
+      ignition::math::Vector3d(_directionX, _directionY, _directionZ));
+  }
+
+  auto lightConfigService = "/world/" + this->dataPtr->worldName +
+    "/light_config";
```

Comment:
Let's run this through `AsValidTopic` just in case the world name has something weird in it